iceoryx_hoofs/source/relocatable_pointer/base_relative_pointer.cpp

```diff
@@ -21,13 +21,13 @@ namespace iox
 {
 namespace rp
 {
-BaseRelativePointer::BaseRelativePointer(ptr_t ptr, id_t id) noexcept
+BaseRelativePointer::BaseRelativePointer(const ptr_t ptr, const id_t& id) noexcept
     : m_id(id)
     , m_offset(computeOffset(ptr))
 {
 }
 
-BaseRelativePointer::BaseRelativePointer(offset_t offset, id_t id) noexcept
+BaseRelativePointer::BaseRelativePointer(const offset_t offset, const id_t& id) noexcept
```

Comment:
@MatthiasKillat @elBoberido 

Here, and in some other place with `Foo(uint64_t, uint64_t)` the clang tidy warning `bugprone-easily-swappable-parameters`. I think this is important to solve as `Foo(1,1)` is legit but not readable and error-prone.

Reply:
I think it is sufficient for now to have only `id_t`. `offset_t` can be changed in the future if necessary.